### ISSUE TITLE
Unable to configure grocy custom component

### ISSUE THREAD

**Issue by ccmy1226** [custom-components/grocy#248]
## Unless all relevant information is provided, I can't help you

After filling in the form for configuring the integration, a simple error message: "Unknown error occurred" is returned.

## Expected behavior
Integration is configured and sensors are available

## To Reproduce

Go to Configuration > Integrations
Click on "+ add integration" and select Grocy
Fill in the required details and click "Submit"
See error

### General information to help debugging:
**What sensors do you have enabled? Are they working and/or what state are they in? Do you have the corresponding functions enabled in Grocy?**

Nothing is enabled yet

**What is your installed versions of Home Assistant, Grocy and this integration?**
Home assistant
Core 2022.9.4
Supervisor 2022.08.6
Host 8.5
Grocy 0.18.2
Custom Grocy: 4.5.2 installed via HACS

**How do you have Grocy installed? Add-on or external?**
Add-on

**Have you added debugging to the log, and what does the log say?**
This error originated from a custom integration.

Logger: custom_components.grocy.config_flow
Source: custom_components/grocy/config_flow.py:42
Integration: Grocy (documentation, issues)
First occurred: 11:19:25 PM (2 occurrences)
Last logged: 11:25:33 PM

Expecting value: line 1 column 1 (char 0)

**JSON service data (if related to using a service)**

**Additional context**
I tried all combination in the configuration panel, all led to the same result.
Localhost and IP address fails right away, my external dns host spins for a while and errors out as well

**Comment by adamschmalhofer** [comment]
Hi ccmy1226!

The TL;DR is upgrade your grocy to 3.3 should fix the problem. 

Your grocy version is very old. Your version is 0.18.2 while 3.3.1 is the current version. While I haven't tested it with a grocy that old, I can reproduce your problem with a grocy 3.1:

Logger: custom_components.grocy.config_flow
Source: custom_components/grocy/config_flow.py:42
Integration: Grocy (documentation, issues)
First occurred: 09:30:39 (1 occurrences)
Last logged: 09:30:39

2 validation errors for SystemInfoDto os field required (type=value_error.missing) client field required (type=value_error.missing)

Hope that helps.
--adaschma

**Comment by grablair** [comment]
Hi @adamschmalhofer, the 0.18.2 version that @ccmy1226 was talking about is the HA Addon version. The underlying Grocy version is different, however.

For example, the [latest version](https://github.com/hassio-addons/addon-grocy/releases) at the time of writing is 0.18.3, but the underlying Grocy version is 3.3.2.

I am experiencing the same issue as @ccmy1226 , and it seems to be due to pygrocy adding a trailing `/` to the API URL:

```
2022-12-15 12:20:39.446 DEBUG (MainThread) [pygrocy.grocy_api_client] generated base url: https://<cloud-endpoint>:443/api/hassio_ingress/cTiwPB15sx7l7K4iKjVkrP6vdOIXAk8BIjlKhrB9bMI/api/
```

If I go to that URL, I do get an error. If I remove the trailing `/`, the call succeeds.

**Comment by grablair** [comment]
@ccmy1226 , I was having the same issue. When I followed these instructions:

> If you have configured the [Grocy add-on](https://github.com/custom-components/grocy#addon) as described, use port 9192 (**without https**).

I had the following values entered:

```
Grocy API URL: http://<internal-ip> (note `http`, as documented in the README)
Grocy API Key: <key>
Port Number: 9192
Verify SSL: No
```

It would fail with the same error as you. When I submitted it _with_ https instead of http, then it succeeded. 🤷🏻‍♂️
